src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -195,131 +201,76 @@ export const MultiSelect = chakra(
-          count++;
-          list.push(currentItem);
-          if (count >= defaultItemsVisible) {
-            break;
-          }
-        }
-        setItemsList(list);
-      }, [items, defaultItemsVisible, setItemsList]);
+      /** Toggle for listOverflow = "expand" */
+      const toggleItemsList = () => {
+        setIsExpandable((prevProp) => !prevProp);
+      };
 
-      // Separate effect for handling listOverflow "expand"
       React.useEffect(() => {
-        if (listOverflow === "expand") {
-          setListHeight("");
-          if (listItemsCount === defaultItemsVisible) {
-            displayDefaultItems();
-          } else {
-            setItemsList(items);
-            setViewAllLabel("View less");
-          }
-        }
-      }, [
-        listOverflow,
-        listItemsCount,
-        items,
-        defaultItemsVisible,
-        displayDefaultItems,
-      ]);
+        setItemsList(isExpandable ? defaultItemsList : items);
+      }, [isExpandable]);
 
-      /**
-       * Render a search input box if the MultiSelect is searchable.
-       * This function conditionally renders a TextInput component for searching options
-       * based on the isSearchable prop. It returns null if the MultiSelect is not searchable.
-       *
-       * @returns {JSX.Element | null} The rendered search input box or null if not searchable.
-       */
-      const showSearchInputBox = () => {
-        if (isSearchable) {
-          return (
-            <TextInput
-              id="multi-select-text-input-id"
-              labelText={`Search ${buttonText}`}
-              isClearable={true}
-              isClearableCallback={clearSearchKeyword}
-              placeholder={`Search for options`}
-              onChange={onChangeSearch}
-              showLabel={false}
-              showRequiredLabel={false}
-              type="text"
-              __css={styles.searchInputBox}
-              marginBottom="s"
-            />
-          );
-        }
-        return null;
+      const ExpandToggleButton = (): JSX.Element => {
+        return (
+          <Button
+            buttonType="text"
+            fontSize="desktop.button.default"
+            id="view-all-text-btn"
+            onClick={toggleItemsList}
+            __css={styles.viewAllButton}
+          >
+            {isExpandable ? "View all" : "View less"}
+          </Button>
+        );
       };
 
-      /**
-       * Generate an array of Checkbox components based on the provided MultiSelectItem.
-       * If the item has children, it creates a Checkbox for the parent and its children.
-       * If the item does not have children, it creates a single Checkbox for the item.
-       * @returns {JSX.Element[]} An array of Checkbox components.
-       */
-      const generateCheckboxArray = (item: MultiSelectItem) => {
+      /** Generate Checkboxe components based on the provided MultiSelectItem. */
+      const MultiSelectItem = ({
```

Comment:
There is nothing wrong with the name of this component but there is a type with the same name, so it can get confusing reading this.